spec/models/book_spec.rb

```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe Book do
+  it { is_expected.to validate_presence_of(:genre) }
+  it { is_expected.to validate_presence_of(:author) }
+  it { is_expected.to validate_presence_of(:title) }
+  it { is_expected.to validate_presence_of(:publisher) }
+  it { is_expected.to validate_presence_of(:year) }
+
+  subject(:book) { build(:book) }
+
+  context 'When the genre is nil' do
```

Comment:
remove all this tests, and add `it { is_expected.to be_valid }` at the beginning (after `subject(:book) { build(:book) }`)
this would validate that the book created by your factory is valid